common/v2/services/Store/StoreProvider.tsx

```diff
@@ -134,6 +138,36 @@ export const StoreProvider: React.FC = ({ children }) => {
     [currentAccounts]
   );
 
+  useEffect(() => {
+    const membershipsItems = Object.values(memberships);
+    if (membershipsItems.length === 0) return;
+    const network = networks.find(({ id }) => DEFAULT_NETWORK === id);
+    if (!network) return;
+    const membershipAccounts = Object.keys(memberships);
+    const unlockProvider = new UnlockProtocolHandler(network);
+    const membershipLookups = R.flatten(
+      membershipAccounts.map(membershipAccount =>
+        memberships[membershipAccount].map(membershipId => {
+          const membershipConfig = MEMBERSHIP_CONFIG[membershipId];
+          return { account: membershipAccount, lockAddress: membershipConfig.contractAddress };
+        })
+      )
+    );
+
+    Promise.all(
+      membershipLookups.map(membershipLookupObj =>
+        unlockProvider.fetchUnlockKeyExpiration(
+          membershipLookupObj.account,
+          membershipLookupObj.lockAddress
+        )
+      )
+    )
+      .then(data => {
+        setMembershipExpiration(Math.max(...data));
```

Comment:
Can we save all expirations and then just extract the max for showing? Not sure if we might need them at some point, also saved all memberships across accounts.

Reply:
I couldn't think of a reason for why we'd need a list of them, but if you think it'll be useful i can add it?

Reply:
Yeah it could possibly be, let's just do it 😄